geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -641,10 +641,22 @@ protected Object processResponse(Message msg) throws Exception {
     }
 
     private void addFunctionException(final FunctionException result) {
-      if (this.functionException == null) {
-        this.functionException = result;
+      if (result instanceof FunctionInvocationTargetException) {
```

Comment:
Can you explain conceptually why we need to wrap these different exception types in a new FunctionException?   Does it have repercussions in other parts of the code if we leave them in their original form (FunctionInvocationTargetException, InternalFunctionInvocationTargetException)?

Reply:
I don't think we should have to, but the product and customers and specific tests were coded to look for the causes() of a functionException and decide whether it should retry or not.  Instead the callers should have been able to just catch the specific exceptions instead of drilling into nested causes. Maybe we can do that for Geode 2.0.

InternalFunctionInvocationTargetException should not be handed back to the caller of the function service.  Those are used internally to decide whether to retry a function or not.  So I can understand not wanting to send those back to a client.